_ap/15/index.adoc

```diff
@@ -0,0 +1,58 @@
+---
+num: 15
+category: "Dynamic Scaling"
+title: "Dynamic Scaling of Data Plane clusters"
+status: "Draft"
+authors:
+  - "Manyanda Chitimbo"
+  - "Steven Hawkins"
+  - "Tom Bentley"
+  - "Peter Braun"
+  - "Ramesh Reddy"
+---
+
+
+== Context and Problem Statement
+
+In the context of both cost management, as well as having the ability to scale to a (virtually) infinite number of manage  instances, 
+our services must have the capability to dynamically scale its underlying infrastructure. 
+This implies both scaling up the infrastructure to accommodate more customers, as well as scaling down the infrastructure to manage costs.
+
+There are a number of things to consider in the context of dynamic scaling:
+. Scaling up/down the number of nodes on an OSD 
+. Provisioning a new OSD
+. Removing an OSD instance
+. Pre-warm nodes to ensure hot capacity for maintaining managed instance provisioning SLOs and SLAs
+
+== Proposed solution
+
+=== Autoscaling of nodes
+
+Cluster Autoscaler will be utilized to allow nodes to be created on demand and removed when they are no longer needed. 
+See https://github.com/kubernetes/autoscaler/tree/master/cluster-autoscaler#introduction[cluster-autoscaler#introduction] for more information.
+
+=== Pre-warming of nodes using Pause pods
+
+In order to ensure that the managed instance provisioning latency is not impacted by the time it takes for a new node to be provisioned, 
+it's possible to reserve hot capacity by configuring overprovisioning: https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/FAQ.md#how-can-i-configure-overprovisioning-with-cluster-autoscaler[how to configure overprovisioning]
+
+
+=== Cluster Full
+
+==== Use node sizing formula
+
+For binpacked solutions, a node sizing formula can be used to determine if a cluster is full or not based on the current number of maximum 
+nodes configured and the current workload. The fleetshard will then propagate the status of the cluster via the managed CR (custom resource) for the control plane
+to observe and act accordingly.
+
+==== Use metrics 
+
+It's also possible to use the exposed cluster autoscaler metrics - see https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/proposals/metrics.md#metrics[cluster autoscaler metrics]
+to determine if the cluster is full or not. The metrics will have to be scraped and stored in Observatorium for the Control Plane to continously 
+poll from and do its logic. This is an advisable pattern for services that do not have a fleetshard nor have concerns over impact of cluster full / over commit upon existing workloads.
```

Comment:
That last sentence is cryptic even to me. Can you elaborate?

Reply:
In kafka, the fleetshard tells us everything. But in some services that do not have the fleetshard operator yet or do not need it (Hypershift service for example) they could rely on polling the metric from Observatorium to see if the cluster is full or not.

It's also a pattern that comes with a lot of risks taking as we approach cluster fullness as it is a reactive model. An example where this reactive model becomes problematic is when the control plane assigns a managed instance `A` in the cluster that's full (but the metric wasn't reported yet). There are cases where this could be an issue to existing workloads if for one reason or the other there are pods restarts which might lead to the existing workload pods competing for resources with the new managed instance `A` that was scheduled when the cluster was full.

Reply:
Right, so take the time to explain that in the doc, so that those other services can weigh up the pros and cons of this approach for them.

Reply:
I've updated the documentation with this info